api/docs_handler_test.go

```diff
@@ -0,0 +1,121 @@
+package api_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/topfreegames/pitaya-admin/api"
+	pt "github.com/topfreegames/pitaya-admin/testing"
+)
+
+func TestGetDocs(t *testing.T) {
+	t.Parallel()
+	if pt.IsConf == false {
+		pt.ConfApp()
+	}
+	a := pt.TestApp
+	var (
+		response    *httptest.ResponseRecorder
+		docsHandler *api.DocsHandler
+	)
+	docsHandler = api.NewDocsHandler(a)
+	tables := map[string]struct {
+		request *http.Request
+		asserts func(response *httptest.ResponseRecorder)
+	}{
+		"test_get_docs_no_param": {
+			request: func() *http.Request {
+				request, err := http.NewRequest("GET", "", nil)
+				assert.NoError(t, err)
+				return request
+			}(),
+			asserts: func(response *httptest.ResponseRecorder) {
+				b := response.Body.String()
+				assert.Equal(t, http.StatusBadRequest, response.Code)
+				assert.Equal(t, `{"success":false, "message":"at least server type must be specified", "reason": "no server type chosen for getting autodoc"}`, b)
+			},
+		},
+		"test_get_docs_by_type_but_sv_does_not_exists": {
+			request: func() *http.Request {
+				request, err := http.NewRequest("GET", "/docs?type=bla", nil)
+				assert.NoError(t, err)
+				return request
+			}(),
+			asserts: func(response *httptest.ResponseRecorder) {
+				b := response.Body.String()
+				assert.Equal(t, http.StatusInternalServerError, response.Code)
+				assert.Equal(t, `{"success":false, "message":"failed to get docs", "reason": "no servers available of this type"}`, b)
+			},
+		},
+		"test_get_docs_success": {
```

Comment:
in success cases can't we validate that we got a valid response? there's no need to test the full response since pitaya should already have this tested, but at least that we got some method's docs and in the protos example, that we got the proto name

Reply:
since i'm starting the same pitaya server for tests, i can test it against the full response and I should've done it early indeed